reduxpractice/src/App.js

```diff
@@ -0,0 +1,56 @@
+import logo from './logo.svg';
+import './App.css';
+import Main from './components/MainComponent'
+import Create from './components/CreateComponent';
+import Delete from './components/DeleteComponent';
+import Show from './components/ShowComponent';
+import {add,loading} from './actions/index'
+import {useDispatch,useSelector} from 'react-redux'
+import { useEffect } from 'react';
+import Update from './components/updateComponent';
+
+function App(props) {
+
+  
+    const dispatch=useDispatch()
+    useEffect(()=>{
+             
+             dispatch(loading(true))
+             fetch('https://reqres.in/api/users?page=2',{method:'GET'}).then((res)=>{
+                   
+                   
+                   res.json().then((res)=>
+                   {
+                           
+                            res.data.forEach(element => {
+                            
+                              //  console.log(element);
+                                dispatch(add(element))    
+                              });
+                           
+                              dispatch(loading(false))
+                        
+                             
+                   })   
+
+             })
+    },[])
+
+    const isLoading=useSelector(state=>state.showReducer.loading)
```

Comment:
if you are selecting values from same reducer, then why two lines?
you could fetch them together e.g
 const {loading: isLoading, showWhat}= useSelector(state => state.showReducer)